### PR TITLE
FIX static-handler middleware for well known files

### PR DESCRIPTION
**Pull request by imalberto**
Appears that static middleware does not quite work for the following: /robots.txt, /crossdomain.xml and /favicon.ico.

This PR addresses this, with unit tests.

### CODE REVIEW

File: lib/app/middleware/mojito-handler-static.js [hunk 1]
```diff
@@ -227,13 +227,13 @@ function staticProvider(store, logger) {
...
-            resource = store.getResource('client', {}, 'asset-ico-favicon');
+            resource = store.getResources('client', {}, 'asset-ico-favicon');
         }
         if (!resource && (path === '/robots.txt')) {
-            resource = store.getResource('client', {}, 'asset-txt-robots');
+            resource = store.getResources('client', {}, 'asset-txt-robots');
         }
         if (!resource && (path === '/crossdomain.xml')) {
-            resource = store.getResource('client', {}, 'asset-xml-crossdomain');
+            resource = store.getResources('client', {}, 'asset-xml-crossdomain');
```
Comment (drewfish) [inline lib/app/middleware/mojito-handler-static.js]: Good catch.  However, `store.getResources()` returns a list of resources, so you might need to `[0]` on it.  Also, I think the `'asset-xml-crossdomain'` argument might instead need to be `{id: 'asset-xml-crossdomain'}`.  (Said another way: I think I rather bungled the previous version of the code :disappointed:  .)

**Comment by rwaldura** [comment]
see also #565 and follow-up #623 which dealt with these special files

**Comment by caridy** [comment]
Aside from those minor details in the tests +1

**Comment by caridy** [comment]
+1

**Pull request merged**